### ISSUE TITLE
DPoP and confidential clients

### ISSUE THREAD

**Issue by gffletch** [danielfett/draft-dpop#114]
I'm wondering if there is any value in adding a section with guidance on how to use DPoP with confidential clients. I'm specifically thinking of the use case of a mobile app that registers via Dynamic Client Registration (DCR) using a pub/priv key pair generated on the mobile device (private key stored in the secure enclave of the device). Once the registration completes, the mobile app instance is effectively a `confidential client`. At that point, should the mobile app generate a new key pair for sender-constraining the access tokens with DPoP (ala FAPI 2) or can it use the same public key used in the DCR registration? This could also apply to server based confidential clients that use PKI for client authentication.

**Comment by selfissued** [comment]
A long time ago I wrote https://datatracker.ietf.org/doc/html/draft-jones-oauth-dpop-implicit-00 to specify how to use DPoP with the implicit flow.  It's out of date and out of sync with the current spec, but if we're going to add to the scenarios that DPoP can be used with, this is another one we should make a conscious decision about.

As a generalization of this, I don't think we currently do a good job saying which response types DPoP can be used with.  We should probably add this to our list of things to be clear on.

**Comment by jogu** [comment]
I think my initial gut feeling would be that for confidential clients the general advice might be to use separate keys as the key rotation mechanisms are different for the two cases and hence the two keys have different expected lifetimes. (i.e. using the dynamic client management endpoint to change the jwks or changing the contents of jwks_uri for the client key, vs requiring reissue of the access token for the DPoP key).

I suspect that general advice may not be applicable to the per-device mobile confidential clients George mentions though.

**Comment by b---c** [comment]
The draft is admittedly light on guidance with respect to keys with DPoP and confidential clients. That's largely because I don't feel like there is clearly obvious guidance to be given. And while I have a similar initial gut feeling as @jogu, I'm extremely hesitant to put speculative gut feeling type guessing advice in a draft that's likely to someday soonish be enshrined as an RFC. People take RFC text seriously and so I try and be conscientious about any advice or guidance given therein.

**Comment by bc-pi** [comment]
The draft is admittedly light on guidance with respect to keys with DPoP and confidential clients. That's largely because I don't feel like there is clearly obvious guidance to be given. And while I have a similar initial gut feeling as @jogu, I'm extremely hesitant to put speculative gut feeling type guessing advice in a draft that's likely to someday soonish be enshrined as an RFC. People take RFC text seriously and so I try and be conscientious about any advice or guidance given therein.

**Comment by selfissued** [comment]
I agree with Brian that we have no definitive guidance to give.  We should close the issue on this basis.
